subsys/net/ip/ipv6.c

```diff
@@ -3777,30 +3930,84 @@ enum net_verdict net_ipv6_process_pkt(struct net_pkt *pkt)
 #if defined(CONFIG_NET_ROUTE)
```

Comment:
maybe all the logic in this code block would need to be put also in another local function with a relevant name. net_ipv6_process_pkt becomes too long now.

Reply:
I will move this new block to new function.